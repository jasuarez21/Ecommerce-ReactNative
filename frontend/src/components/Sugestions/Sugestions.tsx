import React from 'react'
import { ScrollView, SafeAreaView,TouchableHighlight, View, Image, Text, StyleSheet} from 'react-native'

export const Sugestions = () => {
    return (
        <SafeAreaView >
           <Text style={styles.text}>RECOMENDACIONES</Text>
            <ScrollView  horizontal={true}>
                <View style={styles.scroll}>
           <TouchableHighlight>
          <Image
            style={styles.foto}
            source={{
              uri: 'https://cdn.martimotos.com/Martimotos/modules/themeconfigurator/img/5fb2194048bfb04da93ba813a31c913010c1f3b6_minibanneroutletes.jpg?scale.option=fill&w=400&h=0',
            }}
          />
        </TouchableHighlight>
        <TouchableHighlight>
          <Image
            style={styles.foto}
            source={{
              uri: 'https://www.motovery.com/new/wp-content/uploads/2019/04/cascos-moto-scorpion-elche-alicante.jpg',
            }}
          />
        </TouchableHighlight>
        <TouchableHighlight>
          <Image
            style={styles.foto}
            source={{
              uri: 'https://motorline.hu/modules/themeconfigurator/img/9cbb22364e86fea4e977f3de6915ec8a2c67ebba_vespa-ajandek-otletek-min.jpg',
            }}
          />
        </TouchableHighlight>
        <TouchableHighlight>
          <Image
            style={styles.foto}
            source={{
              uri: 'https://www.soymotero.net/sites/default/files/2018-10/Cartel%20Aprilia.jpg',
            }}
          />
        </TouchableHighlight>
        </View>
            </ScrollView>
        </SafeAreaView>
    )
}
const styles = StyleSheet.create({
   text: {
       color: 'white',
       marginTop: 20,
       marginLeft: 20,
       marginBottom: 10,
       borderRadius: 5,
       fontSize: 18,
       borderBottomWidth: 1,
       borderBottomColor: '#FFB800',
       padding: 4
   },
   scroll: {
       display: 'flex',
       flexDirection: 'row',
       width: 870,
       height: 110,
   },
   foto: {
       width: 200,
       borderRadius: 5,
       marginLeft: 10,
       marginTop: 5,
       height: 100
   },
   detail: {
     width: 100,
     height: 100
   }
  });