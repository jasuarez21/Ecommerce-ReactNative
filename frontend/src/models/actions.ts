import { Moto } from "./moto";
import { User } from "./user";

export type MotoActions = 
    | {type: 'GET_MOTOS'; data: Array<Moto>}
    | {type: 'ADD_TO_CART'; moto: Moto}
    | {type: 'LOAD_CART';}
    | {type: 'DELETE_CART'; moto: Moto}
    | {type: 'LOG_IN'; data: User}
    | {type: 'LOG_OUT'; data: User}
    | {type: 'SIGN_UP'; data: User}
    | {type: 'UPDATE_CART'}
    | {type: 'DELETE_ACCOUNT'}
    | {type: 'ERROR'}