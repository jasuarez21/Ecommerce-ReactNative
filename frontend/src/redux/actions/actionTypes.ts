const actionTypes = {
  GET_MOTOS: 'GET_MOTOS',
  LOG_IN: 'LOG_IN',
  LOG_OUT: 'LOG_OUT',
  SIGN_UP: 'SIGN_UP',
  ADD_TO_CART: 'ADD_TO_CART',
  LOAD_CART: 'LOAD_CART',
  DELETE_CART: 'DELETE_CART',
  UPDATE_CART: 'UPDATE_CART',
  SAVE_PRODUCT: 'SAVE_PRODUCT',
  DELETE_ACCOUNT: 'DELETE_ACCOUNT',
  ERROR: 'ERROR'
};

export default actionTypes;
