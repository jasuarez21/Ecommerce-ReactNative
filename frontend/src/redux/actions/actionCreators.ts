import { User } from './../../models/user';
import axios from 'axios';
import {URL_MOTOS, URL_LOGIN, URL_LOGOUT, URL_SIGNUP, URL_DELETE_ACCOUNT, URL_SAVE_PRODUCTS} from '@env';
import actionTypes from './actionTypes';
import {Moto} from '../../models/moto';
import {Dispatch} from 'redux';
import {MotoActions} from '../../models/actions';
import { Order } from '../../models/order';

export function GetMotos() {
  return async (dispatch: Dispatch) => {
      try {   
      const {data} = await axios.get(URL_MOTOS);
      dispatch({
        type: actionTypes.GET_MOTOS,
        data,
      });
    } catch (error) {
      dispatch({
        type: actionTypes.ERROR,
      });
    }
  }
}



export function login(email: string, password: string) {
  return async (dispatch: Dispatch) => {
      try {   
        const {data} = await axios.post(URL_LOGIN, {email, password});
        dispatch({
          type: actionTypes.LOG_IN,
          data,
        });
    } catch (error) {
      dispatch({
        type: actionTypes.ERROR,
      });
    }
  }
}

export function logout() {
  return async (dispatch: Dispatch) => {
      try {   
        const {data} = await axios.post(URL_LOGOUT);
        dispatch({
          type: actionTypes.LOG_OUT,
          data,
        });
    } catch (error) {
      dispatch({
        type: actionTypes.ERROR,
      });
    }
  }
}


export function signUp(email: string, password: string, alias: string, avatar: string) {
  return async (dispatch: Dispatch) => {
      try {   
        const {data} = await axios.post(URL_SIGNUP, {email, password, alias, avatar});
        dispatch({
          type: actionTypes.SIGN_UP,
          data,
        });
    } catch (error) {
      dispatch({
        type: actionTypes.ERROR,
      });
    }
  }
}

export function addToCart(moto: Moto): MotoActions {
  return {
    type: 'ADD_TO_CART',
    moto
  };
}

export function loadCart(): MotoActions {
  return {
    type: 'LOAD_CART'
  };
}

export function deleteFromCart(moto: Moto): MotoActions {
  return {
    type: 'DELETE_CART',
    moto,
  };
}

export function reloadCart(moto: Array<Moto>){
  return async (dispatch: Dispatch) => {
    try {   
      for(let i = 0; i < moto.length; i++){
        await axios.post(`${URL_MOTOS}/${moto[i]._id}`, moto[i]);
        dispatch({
          type: actionTypes.UPDATE_CART
        });
      }
  } catch (error) {
    dispatch({
      type: actionTypes.ERROR,
    });
  }
}
}

export function saveOrder(user: User, cart: Order, token: string){
  user.cart.push(cart);
  const config = { 
    headers: {
        authorization: `Bearer ${token}`,
    }
  }
    return async (dispatch: Dispatch) => {
      try {   
          await axios.post(URL_SAVE_PRODUCTS, {_id: user._id, cart: user.cart}, config);
          dispatch({
            type: actionTypes.SAVE_PRODUCT
          });
    } catch (error) {
      dispatch({
        type: actionTypes.ERROR,
      });
    }
  }
}

export function deleteAccount(user: User, token: string){
    return async (dispatch: Dispatch) => {
      try {   
          await axios.delete(URL_DELETE_ACCOUNT, {headers: {
              authorization: `Bearer ${token}`,
          },
              data:{_id: user._id}
          });
          dispatch({
            type: actionTypes.DELETE_ACCOUNT
          });
    } catch (error) {
      dispatch({
        type: actionTypes.ERROR,
      });
    }
  }
}