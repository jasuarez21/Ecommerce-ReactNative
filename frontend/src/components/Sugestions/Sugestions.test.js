import { render, fireEvent } from '@testing-library/react-native';
import React from 'react';
import ConfigureStore from 'redux-mock-store';
import * as actions from '../../redux/actions/actionCreators';
import { Provider } from 'react-redux';
import {Sugestions} from './Sugestions';

describe('When invoked a Sugestions component', () => {
    test('Should render the sugestion component with inputs', () => {
        const sugestion= render(<Sugestions />)
        expect(sugestion).toMatchSnapshot();
    })
})