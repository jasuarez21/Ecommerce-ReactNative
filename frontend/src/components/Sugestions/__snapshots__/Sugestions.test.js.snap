// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When invoked a Sugestions component Should render the sugestion component with inputs 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <Text
    style={
      Object {
        "borderBottomColor": "#FFB800",
        "borderBottomWidth": 1,
        "borderRadius": 5,
        "color": "white",
        "fontSize": 18,
        "marginBottom": 10,
        "marginLeft": 20,
        "marginTop": 20,
        "padding": 4,
      }
    }
  >
    RECOMENDACIONES
  </Text>
  <RCTScrollView
    horizontal={true}
  >
    <View>
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "height": 110,
            "width": 870,
          }
        }
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Image
            source={
              Object {
                "uri": "https://cdn.martimotos.com/Martimotos/modules/themeconfigurator/img/5fb2194048bfb04da93ba813a31c913010c1f3b6_minibanneroutletes.jpg?scale.option=fill&w=400&h=0",
              }
            }
            style={
              Object {
                "borderRadius": 5,
                "height": 100,
                "marginLeft": 10,
                "marginTop": 5,
                "width": 200,
              }
            }
          />
        </View>
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Image
            source={
              Object {
                "uri": "https://www.motovery.com/new/wp-content/uploads/2019/04/cascos-moto-scorpion-elche-alicante.jpg",
              }
            }
            style={
              Object {
                "borderRadius": 5,
                "height": 100,
                "marginLeft": 10,
                "marginTop": 5,
                "width": 200,
              }
            }
          />
        </View>
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Image
            source={
              Object {
                "uri": "https://motorline.hu/modules/themeconfigurator/img/9cbb22364e86fea4e977f3de6915ec8a2c67ebba_vespa-ajandek-otletek-min.jpg",
              }
            }
            style={
              Object {
                "borderRadius": 5,
                "height": 100,
                "marginLeft": 10,
                "marginTop": 5,
                "width": 200,
              }
            }
          />
        </View>
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Image
            source={
              Object {
                "uri": "https://www.soymotero.net/sites/default/files/2018-10/Cartel%20Aprilia.jpg",
              }
            }
            style={
              Object {
                "borderRadius": 5,
                "height": 100,
                "marginLeft": 10,
                "marginTop": 5,
                "width": 200,
              }
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
